app/services/dependency-resolver.js

```diff
@@ -43,34 +44,50 @@ export default Ember.Service.extend({
   },
 
   resolveAddons: function(addons, dependencies) {
-    let addonPromises = {};
-    Object.keys(addons).forEach((name) => {
-      var value = addons[name];
-
-      addonPromises[name] = this.resolveAddon(name, value);
+    const taskInstance = this.get('resolveAddonsTask').perform(addons, dependencies);
+    return taskInstance.then(() => {
+      return RSVP.resolve(task.value);
```

Comment:
You are correct, and it should be `taskInstance.value`. I was trying to figure out why it still works, and it turns out I never used this value in the `then` where I called `resolveAddons`.